### PR TITLE
DEV-6522 Build fonto-benchmark-runner

### PR DESCRIPTION
**Pull request by mehmetscoskun**

### CODE REVIEW

**Comment by mehmetscoskun** [comment]
I squashed it in the last commit: `DEV-6522 Build fonto-benchmark-runner`

File: README.md [hunk 1]
```diff
@@ -1,2 +1,6 @@
 # fonto-benchmark-runner
-This tool can measure the performance of a function and compare the performances of different functions.
+This tool can measure the performance of a function and compare the performances of different functions in both NodeJS and in the browser. It takes Typescript files as input.
+
+How to use:
+
+- Create files in a performance folder in your project with names ending with .benchmark.ts
```
Comment (bwrrp) [inline README.md:6]: And then run the tool I assume?

File: index.js [hunk 2]
```diff
@@ -0,0 +1 @@
+import server from './src/server.ts';
```
Comment (bwrrp) [inline index.js]: Why is this file here?

Comment (mehmetscoskun) [reply index.js:1]: It should not. But the lib folder is ignored by git. Probably it is there because of one of the previous compiles. I will make sure that the lib folder includes the latest compile in the next commit.

File: src/server.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+	// First: start TSC in watch mode, compiling performance/**/*.ts to some tmp folder
+	// Then: use rollup to bundle that into one file, again in watch mode
+	// Then: host that file
+
+	const fileNames = glob.sync(`${__dirname}/**/*.ts`);
+	const performanceTestFiles = glob.sync(`${cwd}/performance/**/*.benchmark.ts`);
+
+	console.log('server starting');
+
+	// TODO: do this in a watch
```
Comment (bwrrp) [inline src/server.ts:30]: Do we need to fix this? If so, would using a TS rollup plugin (instead of separate TS calls) work?

**Review by bwrrp — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    "glob": "^7.1.6",
+    "node-static": "^0.7.11",
+    "prettier": "^2.0.2",
+    "rollup": "^2.6.1",
+    "slimdom": "^2.3.2",
+    "slimdom-sax-parser": "^1.3.0",
+    "typescript": "^3.8.3"
+  },
+  "devDependencies": {
+    "@rollup/plugin-alias": "^3.1.0",
```
Comment (JosVerburg) [inline package.json]: Why is this a dev dependency? It's used the same way as the other rollup plugins.

File: src/server.ts [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+import commonjs from '@rollup/plugin-commonjs';
+import importBenchmarks from './rollup-plugin-import-benchmarks';
+import ts from 'typescript';
+import * as glob from 'glob';
+
+import * as staticAlias from 'node-static';
+
+import * as rollup from 'rollup';
+
+import alias from '@rollup/plugin-alias';
```
Comment (JosVerburg) [inline src/server.ts]: NIT: Can we move this import to the other plugin imports

File: src/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+		"module": "commonjs",
+		"outDir": "../lib",
+		"sourceMap": false,
+		"target": "esNext",
+		"allowSyntheticDefaultImports": true
+	},
+	"include": [
+		"**/*"
+	]
+}
```
Comment (JosVerburg) [inline src/tsconfig.json]: NIT: No ending newline

File: test/addBenchmark.tests.ts [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+	before(() => {
+		text = '';
+	});
+
+	beforeEach(() => {
+		benchmarkRunner.removeBenchmarks();
+	});
+
+	it('can add a benchmark', async () => {
+		function setup() {
```
Comment (JosVerburg) [inline test/addBenchmark.tests.ts]: Why do we create a setup function and pass it if we don't need and don't test for it. This also goes for a bunch of the other unit tests.

Comment (JosVerburg) [reply test/addBenchmark.tests.ts]: Might be good to add unit tests to check the setup and teardown are executed.

File: test/tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+	},
+	"include": [
+		"../src/**/*",
+		"**/*"
+	],
+	"exclude": [
+		"./assets",
+		"./built"
+	]
+}
```
Comment (JosVerburg) [inline test/tsconfig.json]: NIT: No ending newline

File: tsconfig.json [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+    "module": "commonjs",
+    "moduleResolution": "node",
+    "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true
+  },
+  "include": [
+    "./src/**/*"
+  ]
+}
```
Comment (JosVerburg) [inline tsconfig.json]: NIT: No ending newline

**Review by JosVerburg — APPROVED** [review]

File: test/addBenchmark.tests.ts (hunk 7 above)
Comment (mehmetscoskun) [reply test/addBenchmark.tests.ts]: done

File: test/tsconfig.json (hunk 8 above)
Comment (mehmetscoskun) [reply test/tsconfig.json]: done

File: tsconfig.json (hunk 9 above)
Comment (mehmetscoskun) [reply tsconfig.json]: done

File: src/tsconfig.json (hunk 6 above)
Comment (mehmetscoskun) [reply src/tsconfig.json]: done

File: package.json (hunk 4 above)
Comment (mehmetscoskun) [reply package.json]: Done

File: src/server.ts (hunk 5 above)
Comment (mehmetscoskun) [reply src/server.ts]: Done
